ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/TransactionCommitter.scala

```diff
@@ -215,27 +215,35 @@ private[kvutils] class TransactionCommitter(
               value.getContractState.getContractKey -> Conversions.stateKeyToContractId(key)
           }
 
-        engine
-          .validate(
-            Set(transactionEntry.submitter),
-            SubmittedTransaction(transactionEntry.transaction),
-            transactionEntry.ledgerEffectiveTime,
-            commitContext.participantId,
-            transactionEntry.submissionTime,
-            transactionEntry.submissionSeed,
-          )
-          .consume(
-            lookupContract(transactionEntry, commitContext),
-            lookupPackage(transactionEntry, commitContext),
-            lookupKey(commitContext, knownKeys),
-          )
-          .fold(
-            err =>
-              reject[DamlTransactionEntrySummary](
-                commitContext.recordTime,
-                buildRejectionLogEntry(transactionEntry, rejectionReasonForValidationError(err))),
-            _ => StepContinue[DamlTransactionEntrySummary](transactionEntry)
-          )
+        try {
+          engine
+            .validate(
+              Set(transactionEntry.submitter),
+              SubmittedTransaction(transactionEntry.transaction),
+              transactionEntry.ledgerEffectiveTime,
+              commitContext.participantId,
+              transactionEntry.submissionTime,
+              transactionEntry.submissionSeed,
+            )
+            .consume(
+              lookupContract(transactionEntry, commitContext),
+              lookupPackage(transactionEntry, commitContext),
+              lookupKey(commitContext, knownKeys),
+            )
+            .fold(
+              err =>
+                reject[DamlTransactionEntrySummary](
+                  commitContext.recordTime,
+                  buildRejectionLogEntry(transactionEntry, rejectionReasonForValidationError(err))),
+              _ => StepContinue[DamlTransactionEntrySummary](transactionEntry)
+            )
+        } catch {
```

Comment:
This `catch` needs a test case, which will be done separately to avoid blocking the release further.